umap/views.py

```diff
@@ -773,13 +781,51 @@ def is_unmodified(self):
                 modified = False
         return modified
 
+    def merge(self):
+        """Try to merge conflicts."""
+
+        # Reference data source of the edition.
+        modified_at = self.request.META.get("HTTP_IF_UNMODIFIED_SINCE")
+
+        for name in self.object.get_versions():
```

Comment:
Given that `get_versions()` is reverse ordered by date, why don't we take the first element here? I'm probably missing a use-case with multi-users edit 🤔

Reply:
For merging we need to find **three** versions:

- the current HEAD version (current)
- the one send by POST (candidate)
- the base version of the changes made by the candidate version (reference)

So we can compute the diff between the candidate and its reference, and then see if we can apply this diff on the current HEAD. If yes, bingo, no conflict. Otherwise, back to 422.